src/server/game/Tools/PlayerDump.cpp

```diff
@@ -820,14 +826,13 @@ DumpReturn PlayerDumpReader::LoadDump(std::string const& file, uint32 account, s
     size_t lineNumber = 0;
 
     CharacterDatabaseTransaction trans = CharacterDatabase.BeginTransaction();
-    while (!feof(fin.get()))
+    while (!input.eof())
     {
-        if (!fgets(buf, BUFFER_SIZE, fin.get()))
-        {
-            if (feof(fin.get()))
-                break;
+        input.getline(buf, BUFFER_SIZE);
+        if (input.eof())
+            break;
+        if (input.fail())
             return DUMP_FILE_BROKEN;
-        }
 
         std::string line;
         line.assign(buf);
```

Comment:
All of this could be just
```
std::string line;
while (std::getline(input, line))
{
    // do stuff with line
}
if (input.fail())
    return DUMP_FILE_BROKEN;

// normal success code... think it commits transaction there?
```

Reply:
looks like some nice refactoring to me (that maybe I don't want to include in this PR)

Reply:
You are already killing FILE* and replacing it with as-similar-as-possible looking code using istream (which is not how you should be using istreams)